apis/supernova/src/handler/utils/field-helper/field-utils.js

```diff
@@ -0,0 +1,87 @@
+// eslint-disable-next-line import/no-relative-packages
+import uid from '../../../../../nucleus/src/object/uid';
+import { AUTOCALENDAR_NAME } from '../constants';
+
+/**
+ * Get the field name from the expression.
+ * @param {string} expression
+ * @returns the field
+ */
+export const getField = (expression) => {
+  let exp = expression;
+  exp = exp.trim();
+  if (exp.charAt(0) === '=') {
+    exp = exp.substring(1);
+    exp = exp.trim();
+  }
+  const lastIndex = exp.length - 1;
+  if (exp.charAt(0) === '[' && exp.charAt(lastIndex) === ']') {
+    exp = exp.substring(1, lastIndex);
+    exp = exp.trim();
+  }
+  return exp;
+};
+
+export const findFieldById = (fields, id) => (fields && fields.find((field) => field.qDef?.cId === id)) || null;
+
+export const findLibraryItem = (id, masterItemList) =>
+  (masterItemList && masterItemList.find((item) => item.qInfo.qId === id)) || null;
+
+export const findFieldByName = (name, fieldList) =>
+  (fieldList && fieldList.find((field) => field.qName === name)) || null;
+
+export const initializeId = (field) => ({
+  ...field,
+  qDef: {
+    ...field.qDef,
+    cId: field.qDef?.cId ?? uid(),
+  },
+});
+
+export const initializeField = (field) => ({
+  ...initializeId(field),
+  qOtherTotalSpec: field.qOtherTotalSpec ?? {},
+});
+
+export const setAutoSort = (fields, dimension, self) => {
+  const dim = dimension;
+  fields.forEach((field, index) => {
+    const tags = field.qTags;
+    const sortCriterias = {
+      qSortByLoadOrder: 1,
+    };
+
+    if (typeof self.dimensionDefinition.autoSort === 'function') {
+      self.dimensionDefinition.autoSort(dim, self.properties, tags, sortCriterias, self);
+    } else {
+      // Default auto sorting
+      sortCriterias.qSortByNumeric = 1;
+      sortCriterias.qSortByAscii = 1;
+    }
+
+    if (!dim.qDef.qSortCriterias) {
+      const updatedDimension = { ...dim };
+      updatedDimension.qDef.qSortCriterias = [sortCriterias];
+      Object.assign(dim, updatedDimension);
```

Comment:
I don't understand the usage of `Object.assign` here? Why not just assigned directly on `dim.qDef.qSortCriterias` instead?

Reply:
Umm... true! couldn't remember why I did this... I'll change it.